src/main/java/core/basesyntax/FileWork.java

```diff
@@ -7,38 +7,34 @@
 import java.util.Arrays;
 
 public class FileWork {
-
     private static final String DELIMITERS = "[^A-Za-z0-9]";
     private static final String SPECIFIED_CHARACTER = "w";
+    private static final String SPACE = " ";
 
     public String[] readFromFile(String fileName) {
         File file = new File(fileName);
         StringBuilder builder = new StringBuilder();
         try (BufferedReader bufferedReader = new BufferedReader(new FileReader(file))) {
-            int value = bufferedReader.read();
-            if (value == -1) {
-                return new String[]{};
-            }
-            while (value != -1) {
-                builder.append((char) value);
-                value = bufferedReader.read();
-            }
-            System.out.println(builder.toString());
-            String[] array = builder.toString().toLowerCase().split(DELIMITERS);
-            builder.setLength(0);
-            for (String s : array) {
-                if (s.startsWith(SPECIFIED_CHARACTER)) {
-                    builder.append(s).append(" ");
-                }
-            }
-            if (builder.length() == 0) {
-                return new String[]{};
+            String value = bufferedReader.readLine();
+            while (value != null) {
+                builder.append(value).append(SPACE);
+                value = bufferedReader.readLine();
             }
```

Comment:
you are making the same job twice: appending lines to separate them afterwards. lets process data right when we read the line